vortex-array/src/array/varbin/stats.rs

```diff
@@ -1,129 +1,131 @@
 use std::cmp::Ordering;
 
+use itertools::{Itertools, MinMaxResult};
 use vortex_buffer::Buffer;
 use vortex_dtype::DType;
-use vortex_error::VortexResult;
+use vortex_error::{vortex_panic, VortexResult};
+use vortex_scalar::Scalar;
 
 use crate::accessor::ArrayAccessor;
 use crate::array::varbin::{varbin_scalar, VarBinArray};
 use crate::array::VarBinEncoding;
-use crate::nbytes::ArrayNBytes;
 use crate::stats::{Stat, StatisticsVTable, StatsSet};
-use crate::{ArrayDType, ArrayLen};
+use crate::ArrayTrait;
 
 impl StatisticsVTable<VarBinArray> for VarBinEncoding {
     fn compute_statistics(&self, array: &VarBinArray, stat: Stat) -> VortexResult<StatsSet> {
-        if stat == Stat::UncompressedSizeInBytes {
-            return Ok(StatsSet::of(stat, array.nbytes()));
-        }
-
-        if array.is_empty() {
-            return Ok(StatsSet::default());
-        }
-        array.with_iterator(|iter| compute_stats(iter, array.dtype()))
+        compute_varbin_statistics(array, stat)
     }
 }
 
-pub fn compute_stats(iter: &mut dyn Iterator<Item = Option<&[u8]>>, dtype: &DType) -> StatsSet {
-    let mut leading_nulls: usize = 0;
-    let mut first_value: Option<&[u8]> = None;
-    for v in &mut *iter {
-        if v.is_none() {
-            leading_nulls += 1;
-        } else {
-            first_value = v;
-            break;
-        }
+pub fn compute_varbin_statistics<T: ArrayTrait + ArrayAccessor<[u8]>>(
+    array: &T,
+    stat: Stat,
+) -> VortexResult<StatsSet> {
+    if stat == Stat::UncompressedSizeInBytes {
+        return Ok(StatsSet::of(stat, array.nbytes()));
```

Comment:
I think we did this as a way to propagate uncompressed size from before compression to the compressed array

Reply:
> Then it's even worse, because this is just wrong? You don't know that the offsets array isn't compressed.

duh, the whole point is to get the size when uncompressed

Reply:
> Everyone can ask their children, who eventually know they are PType with a given width and can report back an accurate size

This works for Primitive & Bool but not for VarBinView

Reply:
we still need to store it somehow before the compression

Reply:
@lwwmanning I'm not sure I understand the resolution here?

Reply:
@gatesn resolved because it's way out of scope for this PR, which is meant to optimize VarBin(View) stats. Should have the conversation about UncompressedSizeInBytes elsewhere